### ISSUE TITLE
Fusing layer

### ISSUE THREAD

**Issue by hungnguyen0606** [shekkizh/FCN.tensorflow#11]
According to the paper, we should add a 1x1 convolutional layer on top of pool4 to get a score for each class and use that score to fuse with the final layer in FC32. Finally, we use a devconv layer to get the target image.
However, in your implementation, you convert final layer of FC32, using a devconv layer, to have the same shape with pool4 layer. Then, you directly fuse pool4 with that score.
I just want to know whether the order of these operation matters?

**Comment by shekkizh** [comment]
The order doesn't matter. All that is required is you up-sample(deconv) fc32 to get same size as that of pool4 and add them together .

**Comment by hungnguyen0606** [comment]
Thank you for your answer.

Btw, can you provide the information about your GPU and time to train the network from scratch?

As I'v read in the paper, the authors used up to 7 convolution layers of VGG net, but I observed that you only use 4 layers. Can you provide the performance of your network compared to that of the authors?

**Comment by hungnguyen0606** [comment]
Furthermore, if I want to use this network for other images with different size, should I retrain the network from scratch or only need to resize the label map to the original size?

**Comment by shekkizh** [comment]
The model was trained on a 12gb titanx card. Not sure about exact timing but I believe I trained overnight so ~6-7hrs. 
Not sure what you are referring to as conv layers here. The vgg net model has several conv layers. 

As for input image size, yes it should be possible to do inference with different image size at test time. Just make sure that you do not set any fixed size in placeholders and other places during training.

**Comment by hungnguyen0606** [comment]
Sorry, it's just my misunderstanding about the structure of the network. 
Thank you for your information.
